raft/src/test/java/org/apache/kafka/raft/QuorumStateTest.java

```diff
@@ -952,33 +962,46 @@ public void testVotedToCandidate(KRaftVersion kraftVersion) {
 
     @ParameterizedTest
     @EnumSource(value = KRaftVersion.class)
-    public void testVotedToVotedSameEpoch(KRaftVersion kraftVersion) {
+    public void testNonVoterUnattachedVotedToCandidate(KRaftVersion kraftVersion) {
+        int voter = 1;
+        VoterSet voters = withRemoteVoterSet(IntStream.of(voter), kraftVersion);
```

Comment:
Technically, voter sets with only one voter never send Vote requests. The voter transitions to leader immediately without transitioning to candidate.